### PR TITLE
87-token-renew

### PR DESCRIPTION
**Pull request by meghna-khemka**
# Description

[description]

medic/cht-core [87](https://github.com/medic/cht-release-testing/issues/87)

# Code review checklist

- [X] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [X]  Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [X]  Tested: Unit and/or e2e where appropriate
- [X]  Internationalised: All user facing text
- [X] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: tests/e2e/common/renew-token.wdio-spec.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    await commonPage.goToPeople();
+  });
+
+  it('Refresh page multiple times and verify token gets renewed', async () => {
+    Array.from(Array(3)).forEach(async () => {
+      const beforPageLoadTime = moment();
+      await browser.refresh();
+      const afterPageLoadTime = moment();
+      const ctxExpiry = await getCtxCookieExpiry();
+      chai.expect(ctxExpiry.isBetween(beforPageLoadTime.add(1, 'year'), afterPageLoadTime.add(1, 'year'))).to.be.true;
```
[the remark this answers is not in the document]
Comment (newtewt) [reply tests/e2e/common/renew-token.wdio-spec.js:20]: https://jestjs.io/docs/expect#tobetruthy

File: tests/e2e/common/renew-token.wdio-spec.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+describe('should renew token', async () => {
+
+  beforeEach(async () => {
+    await loginPage.cookieLogin();
+    await commonPage.goToPeople();
+  });
+
+  it('Refresh page multiple times and verify token gets renewed', async () => {
+    Array.from(Array(3)).forEach(async () => {
```
Comment (newtewt) [inline tests/e2e/common/renew-token.wdio-spec.js:15]: The array `forEach` doesn't execute async. Meaning it won't wait for any promises to complete even if you have the async tag.  You might also put in an expect message that indicates what count is set to for future troubleshooting. 

`for(let count = 0; count < 3; count++)`

**Review by newtewt — CHANGES_REQUESTED** [review]
Nice work, I have asked a couple small changes. 

Also, I know this bug was originally raised because the user never logged out after a year and refreshed during the expiry. Another test case, if it's possible, would be to set the browser time forward, refresh, and check the cookie. I believe changing the browser time isn't possible but you might spend a short amount of time seeing if it is.

File: tests/page-objects/common/common.wdio.page.js [hunk 3]
```diff
@@ -76,6 +76,10 @@ const waitForLoaders = async () => {
   });
 };
 
+const getCookies = async (...cookieNameList) => {
```
[the remark this answers is not in the document]
Comment (meghna-khemka) [reply tests/page-objects/common/common.wdio.page.js:79]: Done

File: tests/page-objects/common/common.wdio.page.js [hunk 4]
```diff
@@ -76,6 +76,10 @@ const waitForLoaders = async () => {
   });
 };
 
+const getCookies = async (...cookieNameList) => { 
+  return (await browser.getCookies(cookieNameList));
```
[the remark this answers is not in the document]
Comment (meghna-khemka) [reply tests/page-objects/common/common.wdio.page.js:80]: Done

File: tests/e2e/common/renew-token.wdio-spec.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (meghna-khemka) [reply tests/e2e/common/renew-token.wdio-spec.js:20]: Vanilla Jest doesn't seem to support assertions with a custom error message, hence sticking to use `chai`

File: tests/e2e/common/renew-token.wdio-spec.js (hunk 2 above)
Comment (meghna-khemka) [reply tests/e2e/common/renew-token.wdio-spec.js:15]: done

**Review by newtewt — APPROVED** [review]

**Pull request merged**